ovgs.proto

```diff
@@ -1,175 +1,241 @@
-    bytes public_key_der = 2; // ASN.1 DER encoded
+message GetOwnershipVoucherResponse {
+  bytes voucher_cms = 1; // binary CMS format (rfc5652)
+  bytes public_key_der = 2; // ASN.1 DER encoded
 }
 
 service OwnershipVoucherService {
-    // createGroup creates a group as a child of an existing group
-    rpc createGroup(createGroupRequest) returns (createGroupResponse);
-    // viewGroup returns the domain-certs (keyed by id), serials,
-    // and user/role mappings for that group and all the children.
-    rpc viewGroup(viewGroupRequest) returns (viewGroupResponse);
-    // deleteGroup deletes a named group. Will refuse to do so if there are children groups.
-    rpc deleteGroup(deleteGroupRequest) returns (deleteGroupResponse);
-    // assignUserRole assigns a role to a user in a named group.
-    rpc assignUserRole(assignUserRoleRequest) returns (assignUserRoleResponse);
-    // removeUserRole removes roles of a user in a named group.
-    rpc removeUserRole(removeUserRoleRequest) returns (removeUserRoleResponse);
-    // viewUserRole returns the roles of a user across the groups that they belong to.
-    rpc viewUserRole(viewUserRoleRequest) returns (viewUserRoleResponse);
-    // assignSerial assigns the serial to that group.
-    rpc assignSerial(assignSerialRequest) returns (assignSerialResponse);
-    // unassignSerial removes the serial from the group.
-    rpc unassignSerial(unassignSerialRequest) returns (unassignSerialResponse);
-    // viewSerial returns all the facts about the serial number.
-    rpc viewSerial(viewSerialRequest) returns (viewSerialResponse);
-    // setDomainCert sets the values for the certificate.
-    rpc setDomainCert(setDomainCertRequest) returns (setDomainCertResponse);
-    // readDomainCert reveals the details of the certificate.
-    rpc readDomainCert(readDomainCertRequest) returns (readDomainCertResponse);
-    // deleteDomainCert deletes the cert from the database.
-    rpc deleteDomainCert(deleteDomainCertRequest) returns (deleteDomainCertResponse);
-    // ownershipVoucher issues ownership voucher and returns TPM public key for the serial number
-    rpc ownershipVoucher(ownershipVoucherRequest) returns (ownershipVoucherResponse);
+  // CreateGroup creates a group as a child of an existing group.
+  // Errors will be returned:
+  // ALREADY_EXISTS if group already exists
+  // PERMISSION_DENIED if the user doesn't have access to the parent group
+  rpc CreateGroup(CreateGroupRequest) returns (CreateGroupResponse);
+  // DeleteGroup deletes a named group.
+  // Errors will be returned:
+  // NOT_FOUND if the group doesn't exist
+  // FAILED_PRECONDITION if the parent is set but does not exist
+  // PERMISSION_DENIED if user doesn't have access to parent group
+  rpc DeleteGroup(DeleteGroupRequest) returns (DeleteGroupResponse);
+  // GetGroup returns the domain-certs (keyed by id), serials,
+  // and user/role mappings for that group and all the children.
```

Comment:
Nit: for that group and all the children almost sounds like we have to return everything (domain-certs, serials, user/role mappings) for the entire subtree. Perhaps rephrase to -

 GetGroup returns the domain-certs (keyed by id), serials, user/role mappings and all the child group ids for that group.